tests/conftest.py

```diff
@@ -72,3 +70,12 @@ def spark_session():
     for obj in gc.get_objects():
         if isinstance(obj, Popen) and "pyspark" in obj.args[0]:
             obj.terminate()
+
+
+@pytest.fixture(scope="module")
+def spark_session(spark_session_base):
+    spark = (
+        spark_session_base if spark_session_base else SparkSession.builder.getOrCreate()
```

Comment:
Will happily take advice here on a better way to use the spark_session_base object, this line is mostly to avoid an unused argument lint issue, but perhaps spark_base could be set to autouse at a module level and the spark cleanup would happen automatically at the end of each test module?

Reply:
See [here](https://github.com/quantumblacklabs/kedro/blob/144ffb2032ee1df4654f32881f51987a05c9f14c/tests/conftest.py#L59) and [here](https://github.com/quantumblacklabs/kedro/blob/144ffb2032ee1df4654f32881f51987a05c9f14c/tests/contrib/io/pyspark/test_memory_data_set.py#L56)

Reply:
I think those links just go back to the code this is modifying; the requirement here was to be able to make a custom spark session beyond the vanilla one currently in conftest (specifically `.enableHiveSupport()` and enforcing that the cwd for the duration of that session is in a temporary folder for `metastore_db` folders etc).
Perhaps multiple tests would need different configs so based on a chat with @tolomea we took this approach of a base spark session fixture to keep the cleanup code in one place, and left a vanilla sparksession in conftest for any less fussy spark tests.
I think the important thing is that the base fixture is kicked off before the custom sparksession fixture, hence declaring it as an argument in spark_session (although it is never actually used).
Maybe we just ignore the linter warning and change this simple case back to ```spark = SparkSession.builder.getOrCreate()```?